mathesar_ui/src/systems/table-view/header/header-cell/ColumnHeaderContextMenu.svelte

```diff
@@ -0,0 +1,84 @@
+<script lang="ts">
+  import { ButtonMenuItem } from '@mathesar/component-library';
+  import { iconGrouping, iconSorting } from '@mathesar/icons';
+  import {
+    getTabularDataStoreFromContext,
+    SortDirection,
+    type ProcessedColumn,
+  } from '@mathesar/stores/table-data';
+  import { getSortingLabelForColumn } from './utils';
+
+  export let processedColumn: ProcessedColumn;
+
+  const tabularData = getTabularDataStoreFromContext();
+  $: ({
+    meta: { sorting, grouping },
+  } = $tabularData);
+
+  $: columnId = processedColumn.id;
+
+  $: currentSorting = $sorting.get(processedColumn.id);
+  $: sortingLabel = getSortingLabelForColumn(
+    processedColumn.abstractType.cellInfo.type,
+    !!processedColumn.linkFk,
+  );
+
+  $: hasGrouping = $grouping.hasColumn(columnId);
+
+  function removeSorting() {
+    sorting.update((s) => s.without(columnId));
+  }
+
+  function applySorting(sortDirection: SortDirection) {
+    sorting.update((s) => s.with(columnId, sortDirection));
+  }
+
+  function addGrouping() {
+    grouping.update((g) =>
+      g.withEntry({
+        columnId,
+        preprocFnId: undefined,
+      }),
+    );
+  }
+
+  function removeGrouping() {
+    grouping.update((g) => g.withoutColumns([columnId]));
+  }
+</script>
+
+{#if currentSorting === SortDirection.A}
+  <ButtonMenuItem icon={iconSorting} on:click={removeSorting}>
```

Comment:
Instead of `iconSorting`, can we use `iconSortAscending` and `iconSortDescending` appropriately here?

Reply:
62eda1462

> Also, @rajatvijay I just noticed this small click area which does not trigger the context menu.
> 
> ![image](https://user-images.githubusercontent.com/42411/212115487-7353709f-d5e7-4a74-b9e6-178fda52c27b.png)
> 
> ![image](https://user-images.githubusercontent.com/42411/212115663-863b828c-166e-4270-8aa4-4547f16b2f1b.png)
> 
> Can you fix this so that the context menu click target covers the entire header cell?

fee3337f0

Reply:
This looks good now